fastboot/instance-initializers/content-security-policy.js

```diff
@@ -9,7 +9,9 @@ function readAddonConfig(appInstance) {
   //       if already available through CSP meta element
   assert(
     'Required configuration is available at run-time',
-    addonConfig && addonConfig['reportOnly'] && addonConfig['policy']
+    addonConfig &&
+      Object.prototype.hasOwnProperty.call(addonConfig, 'reportOnly') &&
+      Object.prototype.hasOwnProperty.call(addonConfig, 'policy')
```

Comment:
Wondering if we could assert against the type instead?

```suggestion
    typeof addonConfig === 'object' &&
      typeof addonConfig.reportOnly === 'boolean' &&
      typeof addonConfig.policy === 'string'
```

This seems to be more readable and even make the assertion more strict.

Reply:
I can't seem to find an existing test that actually runs goes through that codepath. Any ideas?

I'm not too experienced with Ember, I have no clue where to begin to setup tests deeply integrated with Fastboot like that.

I can apply your suggestion, but I believe policy should be 'object' right?

Reply:
I changed the implementation as suggested, I will not have time to add tests for this, sorry. Hopefully, this can still be merged and released, as currently, `v2.0.0` crashes on startup when `reportOnly` is set to false in a dev environment.

We're using our fork for now.

Reply:
> The tests are failing with latest changes. Maybe my suggestion isn't working as expected.
> 
> Not sure when I have time to look into it. No one is paying me for maintaining this addon. I'm doing it _after_y regular job.
> 
> If you have time to debug that would be great.
> 
> For tests please have a look at existing Fastboot related tests in `node-tests` folder.

I added a test. It doesn't actually fail with the previous code, since assert only fails in development mode, but as I said I have no clue how to make the test server behave like a dev environment would.

The failing tests were caused by me using `object` for `policy`, I went back to `string` and it works well now.

Reply:
@jelhan Just making sure you've seen this notification. No worries if you didn't have time to get to this yet!